rfcs/90-support-initialization-via-config-string-and-map.md

```diff
@@ -0,0 +1,115 @@
+---
+author: xxchan <xxchan22f@gmail.com>
+status: draft
+updated_at: 2021-06-09
+deprecates:
+  - 13-remove-config-string.md
+---
+
+# GSP-90: Support Initialization Via Config String and Map
+
+## Background
+
+We had config strings like `s3://bucket_name?access_key=xxxx&secret_key=xxxx` before. This is introduced in [GSP-3] but deprecated by [GSP-13].
+
+There are two reasons why config strings are removed, as mentioned in [GSP-13]:
+
+1. Usability: If an application uses [go-storage]'s config string, then it either exposes the format of config string to end user directly (additional burden for end users) or it has to manipulate its config into our format (additional burden for application developers). And if the application wants to combine its other config into a config string, this is also inconvenient.
+2. Implementation: We didn't find a good way to parse pairs from strings easily and safely.
+
+So we let the users of [go-storage] construct pairs directly using `WithXxx()`. 
+
+But passing string config is indeed more convenient, if not much. And actually parsing string into pairs is not that difficult. We can use a registry mechanism like [GSP-48]'s implementation.
+
+## Proposal
+
+So I propose to support service init from config string and map:
+
+We add the following APIs:
+```go
+func NewServicerFromMap(ty string, m map[string]string) (types.Servicer, error) {}
```

Comment:
How about split map support into another PR?

Reply:
Then I prefer to support map first since I plan to use it to support config string...

Reply:
After discussion at [#specs-gsp-90:aos.dev](https://matrix.to/#/#specs-gsp-90:aos.dev), current design is OK to me.

How about just export `NewServicerFromString` to users and don't implement `NewServicerFromMap` for now? (use map internally is LGTM)